### PR TITLE
[RW-9958][risk=no] Convert user confirmation to a remove-all flag

### PR DESCRIPTION
**Pull request by Peter-Lavigne**
**Description**

Converts user confirmation to a remove-all flag by changing the arguments `--authority ALL --remove` into the newly-added `--remove-all` flag. Removes the `ALL` option entirely.

This is my third pass at converting the user confirmation message into Java. The first two attempts can be found [here](https://github.com/all-of-us/workbench/pull/7641).

Future changes that this refactor allows can be found [here](https://docs.google.com/document/d/1-4BPwzEAfxlQqMnF0EAPFmnLOPgp9-fpyMXhPywpgzQ/edit#).

**Testing**

For the same reasons as the previous PR, I have manually tested this change.

To query authorities for a given user run the following SQL: `select * from authority where user_id = (select user_id from user where email = "<email>@fake-research-aou.org");`

1. Remove authorities using `--remove-all`

`./project.rb set-authority --email <user>@fake-research-aou.org --authority DEVELOPER,ACCESS_CONTROL_ADMIN --project $AOU_TEST`

Expected: User has both authorities

`./project.rb set-authority --email <user>@fake-research-aou.org --remove-all --project $AOU_TEST`

Expected: User has no authorities

2. Test for regressions with `--remove`

`./project.rb set-authority --email <user>@fake-research-aou.org --authority DEVELOPER,ACCESS_CONTROL_ADMIN --project $AOU_TEST`

Expected: User has both authorities

`./project.rb set-authority --email <user>@fake-research-aou.org --authority DEVELOPER --project $AOU_TEST --remove`

Expected: User has `ACCESS_CONTROL_ADMIN` authority

3. Validate using both arguments is invalid

`./project.rb set-authority --email <user>@fake-research-aou.org --authority DEVELOPER --project $AOU_TEST --remove-all`

Expected: `ArgumentError`

4. Validate using neither command is invalid

`./project.rb set-authority --email <user>@fake-research-aou.org --project $AOU_TEST`

Expected: `ArgumentError`

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md)
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md)
- [x] If this PR is intended to complete a JIRA story, I have checked that all AC are met for that story
- [x] I have manually tested this change and my testing process is described above
- [x] This PR includes appropriate automated tests, and I have documented any behavior that cannot be tested with code
- [x] If this fixes a bug, ensure the steps to reproduce are in the Jira ticket or provided above.
- [x] I have added explanatory comments where the logic is not obvious
- [x] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented the impacts in the description
- [x] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [x] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [x] If this includes an API change, I have run the relevant E2E tests locally because API changes are not covered by our PR checks

### CODE REVIEW

**Comment by Peter-Lavigne** [comment]
I validated this for `set-authority-local` using the same validations mentioned in the description.

File: api/docs/tools.md [hunk 1]
```diff
@@ -86,31 +88,37 @@ def authority_options(cmd_name, args)
...
+    "--remove-all",
+    ->(opts, _) { opts.remove_all = true},
+    "Removes all authorities from user(s). Exactly one of --authority or --remove-all must be passed."
+  )
   op.add_option(
       "--dry_run",
       ->(opts, _) { opts.dry_run = "true"},
       "Make no changes.")
-  op.add_validator ->(opts) { raise ArgumentError unless opts.email and opts.authority}
+  op.add_validator ->(opts) { raise ArgumentError unless opts.email and ((opts.authority != "") ^ opts.remove_all)}
```
Comment (jmthibault79) [inline api/docs/tools.md:107]: why did you need to add the `!= ""` here? naively, the existing presence check looked fine

Comment (Peter-Lavigne) [reply api/docs/tools.md:107]: The XOR operator doesn't auto-cast strings (`undefined method `^' for "":String (NoMethodError)`) and I was running into errors trying to cast `authority` directly to a boolean like before. And since I was trying to remove our Ruby code if possible, I wasn't particularly interested in researching the exact ruby-ism to describe this.

I'm merging this as-is but I'm open to any suggestions for a follow-up.
